fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/simulation/perception/index/providers/VehicleTree.java

```diff
@@ -76,13 +76,24 @@ public List<VehicleObject> getVehiclesInRange(PerceptionModel searchRange) {
         return vehicleTree.getObjectsInBoundingArea(searchRange.getBoundingBox(), searchRange::isInRange, new ArrayList<>());
     }
 
+    @Override
+    protected VehicleObject addOrGetVehicle(VehicleData vehicleData) {
+        boolean isNew = !indexedVehicles.containsKey(vehicleData.getName());
+        VehicleObject vehicleObject = super.addOrGetVehicle(vehicleData);
+        if (isNew) {
+            vehicleTree.addItem(vehicleObject);
+        }
+        return vehicleObject;
+    }
```

Comment:
This code exists three times now, maybe a better solution could be:

* define a new abstract method `void onVehicleAdded(VehicleObject v)` in class `VehicleIndex` and call this method whenever a vehicle object added to the index.
* implement the method in `VehicleGrid`, `VehicleTree`, ... and add the VehicleObject to the spatial index.

Reply:
Good point! Actually same applies for update and removal operations allowing to only implement `updateVehicles(...)` in the `VehicleIndex` and add the abstract methods `onIndexUpdate` and `onVehicleRemoval`

Reply:
![image](https://user-images.githubusercontent.com/2386865/217876859-b3fe58dc-2de7-4962-8bbd-4bdb1f67c54a.png)